app/api/datastore/v2/applications.rb

```diff
@@ -0,0 +1,40 @@
+module Datastore
+  module V2
+    class Applications < Base
+      include Grape::Kaminari
+
+      version 'v2', using: :path
+
+      resource :applications do
+        desc 'Create an application.'
+        params do
+          requires :application, type: JSON, desc: 'Application JSON payload.'
+        end
+        post do
+          Operations::CreateApplication.new(
+            payload: params[:application]
+          ).call
+        end
+
+        desc 'Return an application by ID.'
+        params do
+          requires :id, type: String, desc: 'Application UUID.'
+        end
+        route_param :id do
+          get do
+            CrimeApplication.find(params[:id]).application
+          end
+        end
+
+        desc 'Return applications with pagination.'
+        params do
+          use :pagination, per_page: 20, max_per_page: 200, offset: false
+        end
+
+        get do
+          paginate(CrimeApplication.all)
```

Comment:
Hmm, shouldn't the pagination be left to the applications integrating with the datastore? (apply/review) to not introduce an additional dependency on gems, etc. and an opinionated way of paginate.

Also not a fan of the metadata coming in the headers.

Reply:
I've removed the grape-kaminari dependency and moved the pagination into the response body.